pkg/registry/networking/networkpolicy/strategy.go

```diff
@@ -46,13 +49,20 @@ func (networkPolicyStrategy) NamespaceScoped() bool {
 func (networkPolicyStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
 	networkPolicy := obj.(*networking.NetworkPolicy)
 	networkPolicy.Generation = 1
+	if !utilfeature.DefaultFeatureGate.Enabled(features.NetworkPolicyEndPort) {
+		dropNetworkPolicyEndPort(networkPolicy)
+	}
 }
 
 // PrepareForUpdate clears fields that are not allowed to be set by end users on update.
 func (networkPolicyStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
 	newNetworkPolicy := obj.(*networking.NetworkPolicy)
 	oldNetworkPolicy := old.(*networking.NetworkPolicy)
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.NetworkPolicyEndPort) && !hasNetworkPolicyEndPort(oldNetworkPolicy) {
```

Comment:
is there a reason for hasNetworkPolicyEndPort check? can we simply drop the field if present as is done in Create? feel like we are iterating twice over all the rules .. perhaps i am overlooking something

Reply:
This was a suggestion from Andrew (as I've based those tests in another feature gate from autoscaling package), but I agree  maybe this is over engineering here.

Anyway this function is used also by the unit tests, will take a look into the impact of removing it.

Reply:
Removed and tests still works, sounds good.